Update dism.rb

back2root opened the pull request:
Ensure that state recognition works with different language versions of Windows

back2root committed:
Update dism.rb

Ensure that state recognition works with different language versions of Windows

back2root committed:
Update dism.rb

Single quote won't work like expected.

back2root committed:
Update dism.rb

Needs to be done in english

back2root committed:
Update dism.rb

Just to be consistent

ferventcoder commented:
@back2root please squash these commits. Perhaps explain in your commit message what this resolves, it is not clear at the moment.

back2root commented:
Hi @ferventcoder,

Sorry, I thought it would be obvious. But here are the explanations for the two types of changes I did:

1. Change:
(Commit: 7b3d4e6, 6f95c4, 8b0cd92)
The script relies on the output of dism. However, as most things within Windows operating system the output from dism is localized within non English versions of Windows. This results into errors e.g. in the process of checking the state of a feature.
I added the parameter "/english" to every call of dism causing dism to generate English output even on non-English versions of Windows.

```
C:\>dism /online /Get-Features
Tool zur Imageverwaltung für die Bereitstellung
Version: 6.3.9600.17031

Abbildversion: 6.3.9600.17031

Featureauflistung für das Paket : Microsoft-Windows-Foundation-Package~31bf3856ad364e35~amd64~~6.3.9600.16384

Featurename : Microsoft-Hyper-V-All
Status : Deaktiviert

Featurename : Microsoft-Hyper-V-Tools-All
Status : Deaktiviert

Featurename : Microsoft-Hyper-V
Status : Deaktiviert
...
```
vs.

```
C:\>dism /english /online /Get-Features
Deployment Image Servicing and Management tool
Version: 6.3.9600.17031

Image Version: 6.3.9600.17031

Features listing for package : Microsoft-Windows-Foundation-Package~31bf3856ad364e35~amd64~~6.3.9600.16384

Feature Name : Microsoft-Hyper-V-All
State : Disabled
```

2. Change : dism "/Source" parameter
(Commit: 7a94564)
It seems that dism does not allow to specify a path within single quotes. Thus I changed it.

ferventcoder commented:
Apologies, the changes are obvious. What lives on are the commit messages. I see you have written messages in the body, but the summary is "Update dism.rb" which isn't very helpful to understanding what was done (most of the commits in this repo update the dism.rb file :D ). 

It seems like you have two commits here. What I would like to see as an end result are two commits, one for the English output and another for the Source parameter change. Have you had an opportunity to create tickets for these changes as well over at https://tickets.puppetlabs.com/browse/modules?

ferventcoder commented:
It seems like you have really just two commits here (hence the request for you to squash and fix up the commit messages).

ferventcoder commented:
For details of what we are looking for with respect to Git commit messages, see https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md#making-changes

glennsarti commented:
Hi @back2root , are you able to squash the commits as fervent coder suggested?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/puppetlabs/puppetlabs-dism?pullRequest=38) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppetlabs-dism?pullRequest=38) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppetlabs-dism?pullRequest=38) it.</sub>
